src/components/NewContainerPull.react.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react/addons';
+import Router from 'react-router';
+import shell from 'shell';
+import containerActions from '../actions/ContainerActions';
+import containerStore from '../stores/ContainerStore';
+
+module.exports = React.createClass({
+  mixins: [Router.Navigation],
+  handleOpenClick: function () {
+    var repo = this.props.pending.repo;
+    if (repo.indexOf('/') === -1) {
+      shell.openExternal(`https://hub.docker.com/_/${this.props.pending.repository}`);
+    } else {
+      shell.openExternal(`https://hub.docker.com/u/${this.props.pending.repository}`);
+    }
+  },
+  handleCancelClick: function () {
+    containerActions.clearPending();
+    this.context.router.transitionTo('new');
+  },
+  handleConfirmClick: function () {
+    containerActions.clearPending();
+    var name = containerStore.generateName(this.props.pending.repo)
+    
+    containerActions.run(name, this.props.pending.repo, this.props.pending.tag);
```

Comment:
I think here we need to transition to view our new container being pulled! Add a line like:

`this.transitionTo('containerHome', {name});`